File: vraptor-blank-project-scala/src/br/com/caelum/vraptor/scala/IndexController.scala
package br.com.caelum.vraptor.scala
import br.com.caelum.vraptor.{Path, Get, Post, Resource};

@Resource
class IndexController {

	@Get @Path(Array("/"))
	def index = new MyModel("It works!")
	
	@Post @Path(Array("/"))
	def index(myModel:MyModel) = myModel
		
	@Get @Path(Array("/jsp"))
	def jsp = new MyModel("It works with JSP too!")
	
	@Get @Path(Array("/list"))
	def list = List("It", "works", "with", "lists")
}

File: vraptor-blank-project-gae/src/main/java/br/com/caelum/vraptor/blank/IndexController.java
package br.com.caelum.vraptor.blank;

import br.com.caelum.vraptor.Resource;
import br.com.caelum.vraptor.Result;

@Resource
public class IndexController {

	private final Result result;

	public IndexController(Result result) {
		this.result = result;
	}

	public void index() {
		result.include("variable", "VRaptor!");
	}
}
